Block/ProductLabel/ProductLabel.php

```diff
@@ -40,38 +41,111 @@ class ProductLabel extends Template implements IdentityInterface
      */
     protected $productLabelCollection;
 
+    /**
+     * @var \Smile\ProductLabel\Model\ImageLabel\Image
+     */
+    protected $imageHelper;
+
+    /**
+     * @var ProductInterface
+     */
+    protected $product;
+
+    /**
+     * @var \Magento\Catalog\Model\ProductRepository
+     */
+    protected $productRepository;
+
+    /**
+     * @var \Magento\Framework\App\RequestInterface
+     */
+    protected $request;
+
     /**
      * ProductLabel constructor.
      *
      * @param \Magento\Backend\Block\Template\Context    $context
      * @param Registry                                   $registry
      * @param ProductLabelOption                         $productLabelCollection
      * @param \Smile\ProductLabel\Model\ImageLabel\Image $imageHelper
+     * @param \Magento\Catalog\Model\ProductRepository   $productRepository
+     * @param \Magento\Framework\App\RequestInterface    $request
      * @param array                                      $data
      */
     public function __construct(
         \Magento\Backend\Block\Template\Context $context,
         \Magento\Framework\Registry $registry,
         ProductLabelOption $productLabelCollection,
         \Smile\ProductLabel\Model\ImageLabel\Image $imageHelper,
+        \Magento\Catalog\Model\ProductRepository $productRepository,
+        \Magento\Framework\App\RequestInterface $request,
         array $data = []
     )
     {
         $this->registry = $registry;
         $this->productLabelCollection = $productLabelCollection;
         $this->imageHelper = $imageHelper;
+        $this->productRepository = $productRepository;
+        $this->request = $request;
 
         parent::__construct($context, $data);
     }
 
     /**
-     * Get Current Catalog Product
+     * Get Current View
      *
-     * @return Product
+     * @return string
+     */
+    public function getCurrentView()
+    {
+        if ($this->request->getControllerName('controller') == 'product') {
+            return 'product';
+        } else {
+            return 'category';
+        }
+    }
+
+    /**
+     * Set Product
+     *
+     * @param ProductInterface $product
+     *
+     * @return $this
+     */
+    public function setProduct(ProductInterface $product)
+    {
+        $this->product = $product;
+        return $this;
+    }
+
+    /**
+     * Get Product By Id
+     *
+     * @param $id
+     *
+     * @return ProductInterface|mixed
+     */
+    public function getProductById($id)
+    {
+        return $this->productRepository->getById($id);
+    }
+
+    /**
+     * Get Current Product
+     *
+     * @return Product|ProductInterface
      */
     public function getCurrentProduct()
     {
-        return $this->registry->registry('current_product');
+        $product = null;
+        if(!empty($this->registry->registry('current_product'))) {
+             $product = $this->registry->registry('current_product');
+        } else {
+            if ($this->product) {
```

Comment:
```if ($this->product)``` then you have the product :) 

You should not reload it with the repository. It's not dramatic on product page, but on category page you will process a load for each product, which is not suitable.

Reply:
if I use the repository I have all the attributes product but with $this->product, I don't have all the attributes.

![data_product](https://user-images.githubusercontent.com/30406381/55145706-8eaef480-5143-11e9-85fd-7e495ad117c1.png)
![product_repository](https://user-images.githubusercontent.com/30406381/55145719-940c3f00-5143-11e9-8d22-8d037ea198fd.png)

Reply:
I know :)

That's exactly the point I wanted you to find. That's because on a listing, Magento is loading only attributes that are set to be "used_in_product_listing". Probably we should add all the attributes that are used for Labels in this case.

The class responsible for this is here : https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Catalog/Model/Config.php#L430

Maybe a plugin on this method will be enough to always have this data loaded properly on a listing page.